Sources/PackageCollectionJSONModel/JSONPackageCollection+v1.swift

```diff
@@ -252,115 +249,66 @@ extension JSONPackageCollectionModel.V1 {
     }
```

Comment:
Thanks so much for adopting this, Yim. I'm doing a test integration with it this morning and one small addition we'd need is a public initialiser for `Compatibility` so we can copy over our data.

```swift
    public init(platform: Platform, swiftVersion: String) {
        self.platform = platform
        self.swiftVersion = swiftVersion
    }
```

Reply:
There are also three other small extensions that I've currently added on our end but which could perhaps also live upstream (but don't have to):

```swift
extension PackageCollectionModel.Platform: Hashable {
    public var hashValue: Int { name.hashValue }
    public func hash(into hasher: inout Hasher) {
        name.hash(into: &hasher)
    }
}


extension PackageCollectionModel.Platform: Comparable {
    public static func < (lhs: Self, rhs: Self) -> Bool {
        lhs.name < rhs.name
    }
}


extension PackageCollectionModel.Compatibility: Comparable {
    public static func < (lhs: Self, rhs: Self) -> Bool {
        if lhs.platform != rhs.platform { return lhs.platform < rhs.platform }
        return lhs.swiftVersion < rhs.swiftVersion
    }
}
```

Reply:
Thanks @finestructure. Amended https://github.com/apple/swift-package-manager/pull/3199/commits/1615db5fa82ece3e3b935e8cdd3b675d8de64498